File: Pattern Printing/1/1_11_21_1211.c
/*

1
1 1
2 1
1 2 1 1
1 1 1 2 2 1
3 1 2 2 1 1

*/
#include <stdio.h>
int main()
{
    int output[50], next_op[50], temp, count = 0, lines, i, j, l, k;
    printf("Enter the number of lines : ");
    scanf("%d",&lines);
    output[0] = 1;
    for(i=0; i<lines; i++)
    {
        for(k=0; output[k]; k++)
            printf("%d ",output[k]);
        l = 0;
        for(j=0,temp = output[0]; j<=k; j++)
        {
            if((temp == output[j]) && (output[j] != '\0'))
                count++;
            else
            {
                next_op[l++] = count;
                next_op[l++] = temp;
                count = 0;
                if(j<k)
                {
                    temp = output[j];
                    j--;
                }
            }
        }
        next_op[l] = '\0';
        for(j=0; j<=l; j++)
            output[j] = next_op[j];
        printf("\n");
    }
    return 0;
}

File: Pattern Printing/1/1_11_21_1211.py
"""

1
1 1
2 1
1 2 1 1
1 1 1 2 2 1
3 1 2 2 1 1

"""
output = []
next_op = []
output.append(1)
count = 0
lines = int(input("Enter the number of lines: "))
for i in range(lines):
    for z in output:
        print(z,end=" ")
    temp = output[0]
    j = 0
    while j < len(output)+1:
        if j < len(output) and temp == output[j]:
            count += 1
            j += 1 
        
        else:
            next_op.append(count)
            next_op.append(temp)
            count = 0
            if j < len(output):
                temp = output[j]
            else:
                break
    output.clear()
    for x in next_op:
        output.append(x)
    next_op.clear()
    print()
